src/rebar_file_utils.erl

```diff
@@ -86,6 +88,36 @@ consult_config(State, Filename) ->
 format_error({bad_term_file, AppFile, Reason}) ->
     io_lib:format("Error reading file ~s: ~s", [AppFile, file:format_error(Reason)]).
 
+-spec symlink_or_create(Source, Target) -> Result when
+    Source :: file:filename(),
+    Target :: file:filename(),
+    Result :: Ok | Error,
+    Ok :: ok,
+    Error :: {error, string()}.
+symlink_or_create(Source, Target) ->
+    SourceExists = ec_file:is_dir(Source) orelse ec_file:is_symlink(Source),
+
+    case SourceExists of
+        true  -> force_link(Source, Target);
+        false -> force_shadow_dir(Target)
+    end.
+
+-spec symlink(Source, Target) -> Result when
+    Source :: file:filename(),
+    Target :: file:filename(),
+    Result :: Ok | Error,
+    Ok :: ok,
+    Error :: {error, string()}.
+symlink(Source, Target) ->
+    case os:type() of
+        {win32, _} ->
+            S = unicode:characters_to_list(Source),
+            T = unicode:characters_to_list(Target),
+            win32_symlink(S, T);
+        _          ->
+            file:make_symlink(Source, Target)
```

Comment:
I think win32_symlink was required in older rebar3 versions. may need to do capability checks.